controllers/manila_controller.go

```diff
@@ -508,27 +509,21 @@ func (r *ManilaReconciler) reconcileNormal(ctx context.Context, instance *manila
 	//
 	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
 	//
-	ospSecret, hash, err := secret.GetSecret(ctx, helper, instance.Spec.Secret, instance.Namespace)
+
+	hash, result, err := ensureSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.Secret},
+		[]string{
+			instance.Spec.PasswordSelectors.Service,
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		manila.NormalDuration,
+	)
 	if err != nil {
-		if k8s_errors.IsNotFound(err) {
-			r.Log.Info(fmt.Sprintf("OpenStack secret %s not found", instance.Spec.Secret))
-			instance.Status.Conditions.Set(condition.FalseCondition(
-				condition.InputReadyCondition,
-				condition.RequestedReason,
-				condition.SeverityInfo,
-				condition.InputReadyWaitingMessage))
-			return manila.ResultRequeue, nil
-		}
-		instance.Status.Conditions.Set(condition.FalseCondition(
-			condition.InputReadyCondition,
-			condition.ErrorReason,
-			condition.SeverityWarning,
-			condition.InputReadyErrorMessage,
-			err.Error()))
-		return ctrl.Result{}, err
+		return result, err
 	}
```

Comment:
Should there be an `else if` here that checks for `result != ctrl.Result{}`, since `ensureSecret` can return a non-empty `ctrl.Result`?

Reply:
In theory yes, but in practice `ensureSecret` is a wrapper that calls `VerifySecret` from lib-common, and the extra `else if` that checks `result != ctrl.Result{}` is part of that function (with the `Condition` update as well).
Do you think we should `unwrap` that part and do that here? I thought that it was also a good idea to keep it in the `ensureSecret` function (in manila_common.go) to not duplicate the same code everywhere.

Reply:
The problem is that you want to return and stop reconciling here if `result != ctrl.Result{}`.  I agree that the `Condition` is already being properly set, but now we need to kick out of the reconcile loop.  See this [1] for an example of the pattern I'm suggesting.

[1] https://github.com/openstack-k8s-operators/keystone-operator/blob/570d08b1749f8139fcf04b7edf2ff905e9fde966/controllers/keystoneapi_controller.go#L718-L734

Reply:
Ok, I think we already had this conversation and now I recall we ended up with the check you mentioned in [1] and [2].
The "price" to pay to have that function is this extra `else if` statement to make sure we get out of the loop. 

[1] https://github.com/openstack-k8s-operators/glance-operator/pull/602
[2] https://github.com/openstack-k8s-operators/glance-operator/pull/603

Reply:
I think I misread your comment. I was looking at the `ensureSecret` function, that has already the `else if` part, while you were looking at the `getConfigSecret()` function. Now I see your point, updating the patch.

Reply:
I think I added the extra check everywhere, including where the `getConfigSecret` function is called.
Let me know if you think I missed something or overlooked any other part.